### DISCUSSION TITLE
Significant increase of Quarkus compiled binary after upgrade from 3.6.7 to 3.7.1

### DISCUSSION THREAD

**Discussion by vincejv** [quarkusio/quarkus#38572] [Community]
Any one noticed a significant increase in the binaries compiled with 3.7.1 compared to 3.6.7? I'm using few Quarkus libraries on top of the framework, mostly just rest client related, and I have observed the docker image published in docker hub increased by 10mb+ (https://hub.docker.com/r/vincejv/qbittorrent-exporter/tags) 0.0.18 (37mb) vs 0.0.17 (28mb).

I thought removing support for older java version would actually reduce the binaries as it will remove legacy code?

Source code: https://github.com/vincejv/qbittorrent-exporter

**Comment by mkouba** [comment]
Do you use Mandrel or GraalVM? 

CC @zakkak

**Reply by gsmet** [reply]
@zakkak could you help us get to the bottom of this one because I don't see any fancy dependencies/extensions?

```
    implementation 'com.google.code.gson:gson:2.10.1'
    implementation 'io.prometheus:simpleclient:0.16.0'
    implementation 'io.micrometer:micrometer-registry-prometheus:1.11.5'

    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
    implementation 'io.quarkus:quarkus-resteasy-reactive'
    implementation 'io.quarkus:quarkus-arc'
    implementation 'io.quarkus:quarkus-config-yaml'
    implementation 'org.jboss.logmanager:log4j2-jboss-logmanager'
```

So if we have a regression, it will probably hit a lot of users.

**Reply by gsmet** [reply]
I created an issue here: https://github.com/quarkusio/quarkus/issues/38600

**Comment by zakkak** [comment]
@vincejv this is happening due to `quarkus.locales` being set to `all` in `application.yaml`. This results in all locales being included in the native image in Quarkus 3.7.1, which was not the case in 3.6.7.

A detailed answer can be found in https://github.com/quarkusio/quarkus/issues/38600#issuecomment-1929738211
